lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -1271,6 +1277,85 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenViewOpToReshape : public OpConversionPattern<AtenViewOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenViewOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Value> sizes;
+    if (!getListConstructElements(op.getSize(), sizes))
+      return op.emitError(
+          "unimplemented: the tensor list is not from list construct");
+
+    auto loc = op.getLoc();
+    ImplicitLocOpBuilder b(loc, rewriter);
+    auto self = adaptor.getSelf();
+    const TypeConverter *typeConverter = getTypeConverter();
+
+    // Convert to the `linalg` types, count the number of negative values,
+    // and determine the product of non-negative values. This lets us compute
+    // the inferred dimensions sizes.
+    auto sizeTy =
+        cast<IntegerType>(typeConverter->convertType(sizes.front().getType()));
+    Value one =
+        b.create<arith::ConstantOp>(sizeTy, rewriter.getIntegerAttr(sizeTy, 1));
+    Value zero =
+        b.create<arith::ConstantOp>(sizeTy, rewriter.getIntegerAttr(sizeTy, 0));
+    Value count = zero;
+    Value knownsize = one;
+    for (auto &size : sizes) {
+      Value convert = typeConverter->materializeTargetConversion(rewriter, loc,
+                                                                 sizeTy, size);
+
+      Value mul = b.create<arith::MulIOp>(knownsize, convert);
+      Value add = b.create<arith::AddIOp>(count, one);
+      Value isneg =
+          b.create<arith::CmpIOp>(arith::CmpIPredicate::slt, convert, zero);
+
+      knownsize = b.create<arith::SelectOp>(isneg, knownsize, mul);
```

Comment:
this makes me feel less dirty than using a mul to do newsize = knownsize + isneg * mul.

should we make a concerted effort to replace, or is it not worth spending time picking up tech debt pennies

Reply:
also , do we have an aten equiv of select op, for onnx to torch lowerings?

also, isneg and knownsize aren't camelCase.

Reply:
Annoyingly we dont have an equivalent. Its a better choice than then the mul-add trick. I have a trick for using the `tensor` variant for scalars (which is the best I can do) but it should get cleaned up.